src/main/java/com/google/devtools/build/lib/skyframe/PlatformLookupUtil.java

```diff
@@ -90,44 +89,44 @@ private static void validatePlatformKeys(
       ImmutableList<ConfiguredTargetKey> platformKeys, Environment env)
       throws InterruptedException, InvalidPlatformException {
     // Load the packages. This should already be in Skyframe and thus not require a restart.
-    Map<ConfiguredTargetKey, PackageIdentifier> targetsToPackageIdentifiers =
-        platformKeys.stream()
-            .distinct()
-            .collect(
-                toImmutableMap(Function.identity(), ctk -> ctk.getLabel().getPackageIdentifier()));
     ImmutableSet<PackageValue.Key> packageKeys =
-        targetsToPackageIdentifiers.values().stream()
+        platformKeys.stream()
+            .map(ConfiguredTargetKey::getLabel)
+            .map(Label::getPackageIdentifier)
             .distinct()
             .map(PackageValue::key)
             .collect(toImmutableSet());
 
-    Map<PackageIdentifier, Package> packages = new HashMap<>();
     Map<SkyKey, ValueOrException<NoSuchPackageException>> values =
         env.getValuesOrThrow(packageKeys, NoSuchPackageException.class);
-    if (env.valuesMissing()) {
-      return;
-    }
+    boolean valuesMissing = env.valuesMissing();
```

Comment:
I don't quite understand why you don't return here if values are missing.

Reply:
This is a source of many Skyframe errors: you have to propagate errors *before* you restart due to missing dependencies, or else the errors will be swallowed and either a) not ever be raised, or b) be raised for the parent of the current skyfunction call. Neither of these is good, and so checking for errors first is essential.

In the simple one-key `env.getValueOrThrow` case this happens automatically, because the `getValueOrThrow` method throws the error immediately and your exception handler catches it, but when requesting multiple keys at once via `env.getValuesOrThrow` it is surprisingly difficult to get it right.

This is why all of my skyfunctions have a lot of error-testing code, because I've been bitten by this too many times **and I still get it wrong the first time**.

Reply:
Ugh. I'll reserve my rant about this for another time. :)